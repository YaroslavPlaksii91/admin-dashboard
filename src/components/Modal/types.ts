import { ReactNode } from 'react';

export type ModalProps = {
  isOpen: boolean;
  title: string;
  children: ReactNode;
  onSubmit: () => void;
  onClose: () => void;
};
