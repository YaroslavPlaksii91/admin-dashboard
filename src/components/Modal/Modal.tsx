import { FC } from 'react';
import { Box, Button, Typography, Modal } from '@mui/material';

import { ModalProps } from './types';

export const ModalComponent: FC<ModalProps> = ({
  isOpen,
  title,
  children,
  onClose,
}) => {
  return (
    <Modal open={isOpen} onClose={onClose} aria-labelledby="modal-title">
      <Box
        sx={{
          position: 'absolute',
          top: '50%',
          left: '50%',
          transform: 'translate(-50%, -50%)',
          padding: 8,
          width: 380,
          textAlign: 'center',
          bgcolor: '#fff',
          outline: 'none',
        }}
      >
        <Typography id="modal-title" variant="h2">
          {title}
        </Typography>

        {children}

        <Button
          variant="text"
          fullWidth
          onClick={onClose}
          sx={{ padding: '8px 0' }}
        >
          Cancel
        </Button>
      </Box>
    </Modal>
  );
};
