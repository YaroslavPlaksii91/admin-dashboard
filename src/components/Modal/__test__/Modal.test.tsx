import { cleanup, render, screen, fireEvent } from '@testing-library/react';

import { ModalComponent } from '../Modal';

describe('Label component', () => {
  afterEach(cleanup);

  const closeModal = jest.fn();
  const title = 'Test Modal';

  it('should be rendered', () => {
    const { asFragment } = render(
      <ModalComponent isOpen={true} title={title} onClose={closeModal}>
        <div>Modal Content</div>
      </ModalComponent>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly', () => {
    render(
      <ModalComponent isOpen={true} title={title} onClose={closeModal}>
        <div>Modal Content</div>
      </ModalComponent>,
    );

    const titleElement = screen.getByText('Test Modal');
    const contentElement = screen.getByText('Modal Content');

    expect(titleElement).toBeInTheDocument();
    expect(contentElement).toBeInTheDocument();
  });

  it('should not be rendered when isOpen false', () => {
    render(
      <ModalComponent isOpen={false} title={title} onClose={closeModal}>
        <div>Modal Content</div>
      </ModalComponent>,
    );

    const titleElement = screen.queryByText('Test Modal');
    const contentElement = screen.queryByText('Modal Content');

    expect(titleElement).not.toBeInTheDocument();
    expect(contentElement).not.toBeInTheDocument();
  });

  it('calls onClose when Cancel button is clicked', () => {
    const { getByText } = render(
      <ModalComponent isOpen={true} title={title} onClose={closeModal}>
        <div>Modal Content</div>
      </ModalComponent>,
    );

    const cancelButton = getByText('Cancel');

    fireEvent.click(cancelButton);

    expect(closeModal).toHaveBeenCalled();
  });
});
