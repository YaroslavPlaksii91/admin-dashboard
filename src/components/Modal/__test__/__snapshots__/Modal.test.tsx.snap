// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label component should be rendered 1`] = `<DocumentFragment />`;

exports[`Modal component should be rendered 1`] = `<DocumentFragment />`;
